crates/subspace-service/src/sync_from_dsn/import_blocks.rs

```diff
@@ -239,69 +245,56 @@ async fn download_and_reconstruct_blocks<PV>(
     segment_index: SegmentIndex,
     piece_provider: &PieceProvider<PV>,
     reconstructor: &mut Reconstructor,
+    dsn_sync_parallelism_level: usize,
 ) -> Result<Vec<(BlockNumber, Vec<u8>)>, sc_service::Error>
 where
     PV: PieceValidator,
 {
-    debug!(%segment_index, "Retrieving pieces of the segment");
+    debug!(%segment_index, %dsn_sync_parallelism_level, "Retrieving pieces of the segment");
 
-    let semaphore = &Semaphore::new(RecordedHistorySegment::NUM_RAW_RECORDS);
+    let semaphore = &Semaphore::new(dsn_sync_parallelism_level);
```

Comment:
This is not a parallelism level at all, this is how many pieces we retrieve from DSN. You should not change it. Concurrency on networking level is already handled with resizable semaphores as a function of number of connections, we don't need extra parameters for this particular use case.